pkg/volume/csi/csi_attacher_test.go

```diff
@@ -331,16 +332,47 @@ func TestAttacherDetach(t *testing.T) {
 		volID      string
 		attachID   string
 		shouldFail bool
+		reactor    func(action core.Action) (handled bool, ret runtime.Object, err error)
 	}{
 		{name: "normal test", volID: "vol-001", attachID: getAttachmentName("vol-001", testDriver, nodeName)},
 		{name: "normal test 2", volID: "vol-002", attachID: getAttachmentName("vol-002", testDriver, nodeName)},
-		{name: "object not found", volID: "vol-001", attachID: getAttachmentName("vol-002", testDriver, nodeName), shouldFail: true},
+		{name: "object not found", volID: "vol-non-existing", attachID: getAttachmentName("vol-003", testDriver, nodeName)},
+		{
+			name:       "volume attachment not found -> success",
+			volID:      "vol-003",
+			attachID:   getAttachmentName("vol-003", testDriver, nodeName),
+			shouldFail: false, //
+			reactor: func(action core.Action) (handled bool, ret runtime.Object, err error) {
+				// return NotFound to all DELETE requests
+				if action.Matches("delete", "volumeattachments") {
+					return true, nil, apierrs.NewNotFound(action.GetResource().GroupResource(), action.GetNamespace())
+				}
+				return false, nil, nil
+			},
+		},
+		{name: "API error", volID: "vol-forbidden", attachID: getAttachmentName("vol-004", testDriver, nodeName)},
```

Comment:
Not sure how this test is different from `object not found` test, also should have more descriptive name if necessary